src/widget/js/app.js

```diff
@@ -244,9 +247,81 @@ window.app = {
     gotLocation(err, location) {
         window.app.state.location = location;
         window.app.gotPieceOfData();
+    },
+    
+    initDetailView: () => {
+      buildfire.deeplink.getData(function (data) {
+        buildfire.datastore.get(window.app.settings.placesTag, function(err, results){
+          if (err) {
+            console.error('datastore.get error', err);
+            return;
+          }
+          let data = results.data;
+          if (data) {
+            window.app.state.bookmarking = data.bookmarking;
+          }
+        });
+
+        if (data) {
+          if (data.data) {
+            let bookmarkData = data.data;
+            window.app.state.bookmarkId = bookmarkData.id;
+            buildfire.datastore.getById(bookmarkData.id, 'places-list', (err, result) => {
+              if (err) console.log(err);
+              let res = [];
+              res = result.data;
+              res.id = result.id;
+              res.distance = bookmarkData.distance;
+              window.app.state.selectedPlace.unshift(res);
+              window.router.navigate(window.app.settings.viewStates.detail);
+              window.app.checkBookmarked(res.id);
+            });
```

Comment:
What are all the code here. We just need to pass place id to navigate to that specific place.